### PR TITLE
Improve mock errors

### PR DESCRIPTION
**Pull request by jonnydgreen**
Fixes: https://github.com/nodejs/undici/issues/787

As suggested, I've added the following improvements to help debug when a mock dispatch is not matched

For a missing path:

```txt
Mock dispatch not matched for path '/wrong': subsequent request to origin http://localhost:9999 was not allowed (net.connect is not enabled for this origin)
Mock dispatch not matched for path '/wrong': subsequent request to origin http://localhost:9999 was not allowed (net.connect is disabled)
```

For a missing method:

```txt
Mock dispatch not matched for method 'WRONG': subsequent request to origin http://localhost:9999 was not allowed (net.connect is not enabled for this origin)
Mock dispatch not matched for method 'WRONG': subsequent request to origin http://localhost:9999 was not allowed (net.connect is disabled)
```

For a missing body:

```txt
Mock dispatch not matched for body 'wrong': subsequent request to origin http://localhost:9999 was not allowed (net.connect is not enabled for this origin)
Mock dispatch not matched for body 'wrong': subsequent request to origin http://localhost:9999 was not allowed (net.connect is disabled)
```

I've also exposed the `mockErrors` as a separate object to `errors` so they are not confused, happy to refactor if you would prefer them to be all together :)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -90,3 +91,4 @@ module.exports.upgrade = makeDispatcher(api.upgrade)
 module.exports.MockClient = MockClient
 module.exports.MockPool = MockPool
 module.exports.MockAgent = MockAgent
+module.exports.mockErrors = mockErrors
```
Comment (jonnydgreen) [inline index.js:94]: I've exposed the `mockErrors` as a separate object to `errors` so they are not confused, happy to refactor if you would prefer them to be all together :)

**Review by mcollina — APPROVED** [review]
Good work!

I think we should add some check/assertions on Headers as well. I just realised they are not supported.

**Comment by jonnydgreen** [comment]
Thanks! Just to clarify, when you say check/assertions on Headers as well, do you mean match on mock dispatch headers? For example:

```js
mockPool.intercept({
  path: '/foo',
  method: 'GET',
  headers: ['hello', 'there'] // Only a request for this origin,path,method,headers will be intercepted
}).reply(200, 'foo')
```

If so, I can do that in a separate PR over the weekend if that works for you? :)

**Comment by mcollina** [comment]
That's perfect thanks.

**Pull request merged**